output/out_logzio.go

```diff
@@ -81,27 +88,46 @@ func FLBPluginRegister(ctx unsafe.Pointer) int {
 // If the plugin reports an error, the engine will not load the instance.
 //export FLBPluginInit
 func FLBPluginInit(ctx unsafe.Pointer) int {
-	if err := initConfigParams(ctx); err != nil {
-		logger.Log(fmt.Sprintf("failed to initialize output configuration: %v", err))
-		plugin.Unregister(ctx)
+	if ctx != nil {
+		if err := initConfigParams(ctx); err != nil {
+			logger.Debug(fmt.Sprintf("failed to initialize output configuration: %v", err))
+			plugin.Unregister(ctx)
+			return output.FLB_ERROR
+		}
+
+		output.FLBPluginSetContext(ctx, output.FLBPluginConfigKey(ctx, "id"))
+	} else {
 		return output.FLB_ERROR
 	}
 	return output.FLB_OK
 }
 
+//export FLBPluginFlush
+func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
+	return output.FLB_OK
+}
+
 // FLBPluginFlush Upon flush time, when Fluent Bit wants to flush it's buffers,
 // the runtime flush callback will be triggered.
 // The callback will receive the configuration context,
 // a raw buffer of msgpack data,
 // the proper bytes length and the associated tag.
 // When done, there are three returning values available: FLB_OK, FLB_ERROR, FLB_RETRY.
-//export FLBPluginFlush
-func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
+//export FLBPluginFlushCtx
+func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int {
 	var ret int
 	var ts interface{}
 	var record map[interface{}]interface{}
+	var id string
+	if ctx != nil {
+		id = output.FLBPluginGetContext(ctx).(string)
+	}
 
-	// Create Fluent-Bit decoder
+	if id == "" {
```

Comment:
Is it mandatory to set id? what happens if someone not set it ( for backward compatibility )